### PR TITLE
Add dashboard image in ElasticsearchVersion

### PR DESCRIPTION
**Pull request by raihankhan**

### CODE REVIEW

**Review by kamolhasan — CHANGES_REQUESTED** [review]
Perform `make gen fmt`, and push again.

File: apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go [hunk 1]
```diff
@@ -241,76 +217,35 @@ func (ed *ElasticsearchDashboard) ClientCertificateCN(alias ElasticsearchDashboa
...
 }
 
-func isOfficialTypes(group string) bool {
-	return !strings.ContainsRune(group, '.')
-}
+func (ed *ElasticsearchDashboard) GetPhaseFromCondition() DashboardPhase {
 
-func (ed *ElasticsearchDashboard) PatchStatus(ctx context.Context, r client.Client, obj client.Object, transform clientutil.TransformFunc, opts ...client.PatchOption) (client.Object, kutil.VerbType, error) {
+	var phase DashboardPhase
+	phase = DashboardPhaseProvisioning
```
Comment (kamolhasan) [inline apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go:226]: phase := DashboardPhaseProvisioning

File: apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go [hunk 2]
```diff
@@ -241,76 +217,35 @@ func (ed *ElasticsearchDashboard) ClientCertificateCN(alias ElasticsearchDashboa
...
 
-	key := types.NamespacedName{
-		Namespace: obj.GetNamespace(),
-		Name:      obj.GetName(),
-	}
-	err := r.Get(ctx, key, obj)
-	if err != nil {
-		return nil, kutil.VerbUnchanged, err
+	if kmapi.IsConditionFalse(ed.Status.Conditions, string(DashboardConditionAcceptingConnection)) &&
```
Comment (kamolhasan) [inline apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go:228]: Check, !True() istead of isFalse()

File: apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go [hunk 3]
```diff
@@ -241,76 +217,35 @@ func (ed *ElasticsearchDashboard) ClientCertificateCN(alias ElasticsearchDashboa
...
-		}
-	}
-	return false
+	return fmt.Sprintf("%s-%s", databaseName, ed.GetCertSecretName(DefaultElasticsearchClientCertAlias))
 }
 
-func isOfficialTypes(group string) bool {
-	return !strings.ContainsRune(group, '.')
-}
+func (ed *ElasticsearchDashboard) GetPhaseFromCondition() DashboardPhase {
```
Comment (kamolhasan) [inline apis/dashboard/v1alpha1/elasticsearchdashboard_helpers.go:223]: Move this function to `pkg/phase`. Make it generic for all Dashboards.

File: apis/dashboard/v1alpha1/elasticsearchdashboard_webhook.go [hunk 4]
```diff
@@ -163,56 +146,35 @@ func (r *ElasticsearchDashboard) Validate() error {
 	// suffixes(max 13 characters) to resources that it creates and watches.  Therefore, ElasticsearchDashboard
 	// names must have length <= 63-13=50. If we don't validate this here,
 	// then ElasticsearchDashboard resource creations will fail later.
-	if len(r.Name) == validation.DNS1035LabelMaxLength-13 {
-		allErr = append(allErr, field.Invalid(field.NewPath("Name"), r.Name,
-			fmt.Sprintf("%v is too long. keep it within 50 characters", r.Name)))
+	if len(ed.Name) == validation.DNS1035LabelMaxLength-13 {
```
Comment (kamolhasan) [inline apis/dashboard/v1alpha1/elasticsearchdashboard_webhook.go:149]: Give this responsibility to k8s itself

File: pkg/phase/phase.go [hunk 5]
```diff
@@ -17,11 +17,43 @@ limitations under the License.
...
+	dapi "kubedb.dev/apimachinery/apis/dashboard/v1alpha1"
 	api "kubedb.dev/apimachinery/apis/kubedb/v1alpha2"
 
 	kmapi "kmodules.xyz/client-go/api/v1"
 )
 
+func DashboardPhaseFromCondition(conditions []kmapi.Condition) dapi.DashboardPhase {
+
+	phase := dapi.DashboardPhaseProvisioning
+
```
Comment (kamolhasan) [inline pkg/phase/phase.go:29]: if !IsConditionTrue(Provisioned){
return provisioning 
}

if !IsConditionTrue(AcceptingConnection){
return NotReady
}

if

File: apis/dashboard/v1alpha1/elasticsearchdashboard_webhook.go (hunk 4 above)
Comment (raihankhan) [reply apis/dashboard/v1alpha1/elasticsearchdashboard_webhook.go:149]: done

File: pkg/phase/phase.go (hunk 5 above)
Comment (raihankhan) [reply pkg/phase/phase.go:29]: done

**Review by kamolhasan — DISMISSED** [review]
LGTM!

**Review by raihankhan — COMMENTED** [review]
ok

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Pull request merged**